pkg/stub/handler.go

```diff
@@ -109,3 +118,50 @@ func (h *Handler) syncIngressUpdate(ci *ingressv1alpha1.ClusterIngress) error {
 
 	return nil
 }
+
+func (h *Handler) deleteIngress(ci *ingressv1alpha1.ClusterIngress) error {
+	ds, err := h.manifestFactory.RouterDaemonSet(ci)
+	if err != nil {
+		return fmt.Errorf("couldn't build DaemonSet object for deletion: %v", err)
+	}
+	err = sdk.Delete(ds)
+	if err != nil {
+		logrus.Infof("Error deleting DaemonSet %s", ds.Name)
+		return err
+
+	}
+	return nil
+}
+
+// getDaemonSetOwnerRef returns an object as OwnerReference
+func getDaemonSetOwnerRef(ds *appsv1.DaemonSet) metav1.OwnerReference {
+	d := &appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DaemonSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ds.Name,
+			Namespace: ds.Namespace,
+		},
+	}
+
+	err := sdk.Get(d)
+	if err != nil {
+		logrus.Infof("Couldn't get DaemonSet %s Namespace %s : %s", ds.Name, ds.Namespace, err)
```

Comment:
This is one of the places where instead of creating &appsv1.DaemonSet{} and calling sdk.Get(), using kubeClient directly would have been simpler! I'm fine with this.

Reply:
i would like to stick using the sdk, if we need to use the k8s api directly we should fix it on a PR for the sdk